mathesar/api/exceptions/mixins.py

```diff
@@ -0,0 +1,152 @@
+
+        """
+        This method build on top of `build_pretty_errors` method of the superclass
+        It provides the following additional features
+        1. Avoids processing prettified exceptions
+        2. Add field to the pretty exception body if raised by field validation method
+        """
+        pretty = []
+        for error_type in errors:
+            error = errors[error_type]
+            if error_type == 'non_field_errors':
+
+                if self.is_pretty(error):
+                    pretty.append(error)
+                else:
+                    pretty.extend(self.get_non_field_error_entries(errors[error_type]))
+            else:
+                field = self.fields.fields[error_type]
+                if isinstance(field, Serializer) and type(errors[error_type]) == dict:
+                    field.initial_data = self.initial_data[error_type]
+                    child_errors = field.build_pretty_errors(errors[error_type])
+                    pretty += child_errors
+                    continue
+                if self.is_pretty(error):
+                    if 'field' not in error or error['field'] is None or str(error['field']) == 'None':
+                        error['field'] = error_type
+                    pretty.append(error)
+                else:
+                    pretty.extend(self.get_field_error_entries(errors[error_type], field))
+        if pretty:
+            return pretty
+        return []
+
+    def _run_validator(self, validator, field, message):
+        """
+        This method build on top of `_run_validator` method of the superclass
+        It provides the following additional features
+        1. Includes serializer if `required_context` is True similar to the behaviour of drf
+        """
+        try:
+            args = []
+            if getattr(validator, 'requires_context', False):
+                args.append(field)
+            validator(self.initial_data[field.field_name], *args)
+        except (DjangoValidationError, RestValidationError) as err:
+            err_message = err.detail[0] \
+                if hasattr(err, 'detail') else err.message
+            return err_message == message
+
+    @property
+    def errors(self):
+        """
+        This method build on top of `errors` property of the superclass to return a list instead of a dictionary
+        """
+        ugly_errors = super(FriendlyErrorMessagesMixin, self).errors
+        pretty_errors = self.build_pretty_errors(ugly_errors)
+        return ReturnList(pretty_errors, serializer=self)
+
+    @property
+    def field_map(self):
+        """
+        This method build on top of `field_map` property of the superclass
+        It provides the following additional features
+        1. Adds `ListSerializer` to `relation` field list
+        """
+        return {
+            'boolean': ['BooleanField', 'NullBooleanField'],
+            'string': ['CharField', 'EmailField', 'RegexField', 'SlugField',
```

Comment:
If you're splitting a list or tuple into multiple lines, I much prefer this pattern to having a few listed on every line:
```
 'string': [
    'CharField',
    'EmailField',
    'RegexField',
    'SlugField',
    'URLField',
    'UUIDField',
    'FilePathField',
]

Reply:
Also applies to everywhere else.